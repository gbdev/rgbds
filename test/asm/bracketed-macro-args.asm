MACRO printargs
	PRINTLN "first = \<1>"
	FOR I, 2, _NARG
		PRINTLN "next = \<{d:I}>"
	ENDR
	PRINTLN "last = \<{d:_NARG}>"
ENDM

	printargs A, B, C, D

MACRO mac
	println \<2__> + \<1_2> + \<\1>
	def x = 2
	println \<{d:x}> + \<1_{d:x}> + \<\<\<13>>>
	def y equs "NARG"
	println \<x> + \<1_{d:x}_> + \<\<\<_{y}>>>
ENDM

	mac 10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 110, 120, 1
