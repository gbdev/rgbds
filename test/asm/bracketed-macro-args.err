error: bracketed-macro-args.asm(33) -> bracketed-macro-args.asm::bad(26):
    Bracketed symbol "nonnumeric" is not numeric
error: bracketed-macro-args.asm(33) -> bracketed-macro-args.asm::bad(27):
    Invalid bracketed macro argument '\<0>'
error: bracketed-macro-args.asm(33) -> bracketed-macro-args.asm::bad(28):
    Bracketed symbol "undefined" does not exist
error: bracketed-macro-args.asm(33) -> bracketed-macro-args.asm::bad(29):
    Macro argument '\<2>' not defined
error: bracketed-macro-args.asm(33) -> bracketed-macro-args.asm::bad(30):
    Macro argument '\<2>' not defined
warning: bracketed-macro-args.asm(39) -> bracketed-macro-args.asm::toolong(36): [-Wlong-string]
    Bracketed symbol name too long
error: bracketed-macro-args.asm(39) -> bracketed-macro-args.asm::toolong(36):
    Bracketed symbol "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstu" does not exist
error: Assembly aborted (6 errors)!
