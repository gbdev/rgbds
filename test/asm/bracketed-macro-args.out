first = A
next = B
next = C
last = D
$F0
$F0
$F0
nonnumeric
zero
undefined
two
2

